### DISCUSSION TITLE
Anyone know the piece of code that suggests exporting the variables to the startup file

### DISCUSSION THREAD

**Discussion by Maxrovr** [Homebrew/discussions#3226] [Getting started]
When brew installs a package, it spits out a echo export statement the user can run that adds the package to the path and provides the right startup file into which the export needs to be echoed into. I'm curious to know how this has been implemented. Is it a shell to startup file mapping or is there something much clever happening?

**Comment by gromgit** [comment]
Neither, the instructions are hardcoded in a `caveats` section of whichever formula files require them, since most formulae _don't_ need such instructions. See, for instance, https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/coreutils.rb.
